File: 30 days of code/# Java/Day 11 - 2D arrays.java
/**
 *              2D ARRAYS | JAVA
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 11
 */

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws FileNotFoundException {
        File file = new File("input.txt");
        Scanner scan = new Scanner(file);
        int n = 6;
        int[][] matrix = new int[n][n];
        for(int i = 0; i < n; ++i) {
            for (int j = 0; j < n; ++j) {
                matrix[i][j] = scan.nextInt();
            }
        }
        scan.close();

        hourGlass(matrix, n);
    }

    private static void hourGlass(int[][] matrix, int n) {
        int max = Integer.MIN_VALUE;
        for(int i = 1; i < n - 1; ++i) {
            for (int j = 1; j < n - 1; ++j) {
                int currentSum = matrix[i - 1][j - 1] + matrix[i - 1][j] + matrix[i - 1][j + 1]
                                                        + matrix[i][j] +
                                 matrix[i + 1][j - 1] + matrix[i + 1][j] + matrix[i + 1][j + 1];
                max = (currentSum > max) ? currentSum : max;
            }
        }

        System.out.println(max);
    }
}

File: 30 days of code/__ C#/Day 11 - 2D arrays.cs
/**
 *              2D ARRAYS | C#
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 11
 */

using System;
using System.IO;

class Program
{
    //static TextReader read = Console.In;
    static StreamReader read = new StreamReader("input.txt");

    static void Main(string[] args)
    {
        int n = 6;
        int[][] matrix = new int[6][];
        for (int i = 0; i < n; ++i)
        {
            string[] line = read.ReadLine().Split(' ');
            matrix[i] = Array.ConvertAll(line, int.Parse);
        }
        read.Close();

        HourGlass(matrix, n);
    }

    private static void HourGlass(int[][] matrix, int n)
    {
        int max = int.MinValue;
        for (int i = 1; i < n - 1; ++i)
        {
            for (int j = 1; j < n - 1; ++j)
            {
                int currentSum =
                    matrix[i - 1][j - 1] + matrix[i - 1][j] + matrix[i - 1][j + 1]
                                           + matrix[i][j] +
                    matrix[i + 1][j - 1] + matrix[i + 1][j] + matrix[i + 1][j + 1];
                max = (currentSum > max) ? currentSum : max;
            }
        }

        Console.WriteLine(max);
    }
}